src/mgr/PyModule.h

```diff
@@ -0,0 +1,120 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 John Spray <john.spray@redhat.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ */
+
+#pragma once
+
+#include "Python.h"
+#include "Gil.h"
+
+#include <string>
+#include "common/Mutex.h"
+#include <memory>
+
+std::string handle_pyerror();
+
+/**
+ * A Ceph CLI command description provided from a Python module
+ */
+class ModuleCommand {
+public:
+  std::string cmdstring;
+  std::string helpstring;
+  std::string perm;
+
+  // Call the ActivePyModule of this name to handle the command
+  std::string module_name;
+};
+
+class PyModule
+{
+  mutable Mutex lock{"PyModule::lock"};
```

Comment:
might need use a different name for every instance of `PyModule`, otherwise lockdep will be annoyed. see http://pulpito.ceph.com/kchai-2018-01-07_09:47:26-rados-wip-kefu-testing-2018-01-07-1345-distro-basic-mira/2039373/

Reply:
Hmm, I'm not sure if this is because of this lock: we're getting a floating point exception in lockdep_register rather than a proper lockdep error.  The backtrace looks like it's happening somewhere being called back from the python code during import (possibly construction of librados/libcephfs bits?).  I'm investigating.